test/test_deferral.rb

```diff
@@ -146,4 +146,17 @@ def close
     assert r1.closed?
     assert r2.closed?
   end
+
+  test 'pass arguments to defer' do
+    r1 = Resource.new
+    r2 = Resource.new
+    1.times do
+      r = r1
+      Deferral.defer(r){|r| r.close }
+      r = r2
+      Deferral.defer(resource: r){|resource:| r.close }
```

Comment:
It seems that the variable name in deferred block should be `resource`, doesn't it?